rfcs/0016-permissionless-assets-for-ibc-and-cw.md

```diff
@@ -0,0 +1,217 @@
+
+The chain needs to make sure that permissionless assets can be transferred to accounts that do not have ED in parachain's native token. It is also required to fail transfers that will result in drop of native token below ED if this account has nonsufficient tokens. [Reference](https://substrate.stackexchange.com/questions/2447/influence-of-existential-deposits-on-account-assets)
+
+
+#### Recieving Account doesnt hold pica during bridging
+**Potential solution**: relayer will send PICA to the reciever in case reciever doesnt have it. In case bridged asset is FlatFee asset, additional fee equal to ED in PICA will be charged from the transferred amount. Otherwise, percentage fee will be applied(To be configured)
+
+#### orml-tokens pallet EDs
+Currently assets are being stored in `orml-tokens` pallet which require Existential Deposits for stored assets. 
+
+**Potential Solution**: for non sufficient assets still provide non-zero Existential Deposit parameter which is similar to minimum balance in Asset Hub implementation. Then the Exitstential logic for orml-tokens pallet will stay the same because account wont be able to hold a non sufficient asset without ED in parachain's native token. This will require a new field in AssetsRegistry: is_sufficient. All permissionless assets will have this field to be false by default. Assets created via permissioned extrinsic will allow to choose if asset is sufficient or not. Asset's sufficiency can be changed by governance.
+
+### Scenarios
+
+#### On chain transactions
+
+CW and PD.js permissionless assets are same, because anybody can upload CW contract and execute any transaction on it.
+
+
+**Create**
+- No requires inputs
+- `CreationFee` is payed in any assets, default to PICA. Or optional `FeeAsset` parameter is provided with sufficient asset identifier.  
+- `Metatdata` properties are optional. 
+- No other parameters.
+- Optional `admin`s, which defaults to CW address or origin if not specified. 
+
+#### Bridges
+
+Each bridge has `origin`.
+In case of IBC it is `channel and port`.
+In case of XCM it is `MultiLocation origin`.
+Any such origin can be hashed (to account) to produce Account32 which may be set be considered `mint_admin` of asset.
+
+Via governance additional PD.js account or CW contract can be made to be second `metadata_owner` of asset. 
+
+**Create**
+- Happens automatically on transfer [1].
+- When assets are created by other chains over bridges, no `Metadata` or `sufficiency` provided
+- We know `location` of assets from origin and received asset id and this parameter is not optional.
+- Location is mapped to `network_id` and `bridge` type to `protocol_id`.
+- We take `CreationFee` from relevant source described per bridge next.
+
+#### Sufficiency of IBC ICS-020
+
+1. New token arrives.
+2. Relayer pays for execution, including `CreationFee` in PICA.
+3. Relayer pays fee for ED for account creation in PICA.
```

Comment:
I fixed CreationFee.

Reply:
if you refer paying ED for new accounts not in PICA for new IBCed non sufficient assets accunts. I do not know.

Reply:
**Note** Nothing changes in case of sufficient assets, all bridge transfers were and are payed in transferred asset when fee payed as part of it.  cc @josietyleung

Reply:
@kkast relayer just usual transaction.

Reply:
my wallet can relay.

Reply:
so like usuall wallet is taken away amounts when it does extrinsics, same way relayer operates. this just like usual substrate. no magic. just transfer from user wallet which is relayer to some other account or fail tx of packet delivery.